src/configurationProvider.ts

```diff
@@ -0,0 +1,74 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { parse } from 'jsonc-parser';
+import { OmniSharpServer } from './omnisharp/server';
+import * as serverUtils from './omnisharp/utils';
+import { AssetGenerator, addTasksJsonIfNecessary, createLaunchConfiguration, createAttachConfiguration, containsDotNetCoreProjects} from './assets';
+import { WorkspaceInformationResponse } from './omnisharp/protocol';
+
+export class CSharpConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private server: OmniSharpServer;
+
+    public constructor(server: OmniSharpServer) {
+        this.server = server;
+    }
+
+    /**
+     * TODO: Remove function. 
+     * 
+     * Note: serverUtils.requestWorkspaceInformation only retrieves one folder for multi-root workspaces. Therefore, generator will be incorrect for all folders
+     * except the first in a workspace. 
+     */
+    private checkWorkspaceInformationMatchesWorkspaceFolder(folder: vscode.WorkspaceFolder, info: WorkspaceInformationResponse): boolean {
+        return info && ((info.MsBuild && info.MsBuild.SolutionPath === folder.uri.fsPath) || (info.DotNet && info.DotNet.RuntimePath === folder.uri.fsPath));
+    }
+
+    /**
+	 * Returns a list of initial debug configurations based on contextual information, e.g. package.json or folder.
+	 */
+    provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.DebugConfiguration[]> {
```

Comment:
>undefined [](start = 64, length = 9)

Can this really be undefined? If so, we don't seem to handle it.

Reply:
Whoops I did see the question, VsCode had the function defined as that in their interface and I copied it over. So I added a check for ```folder``` but it got removed during other changes.